packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/Actions.tsx

```diff
@@ -292,195 +298,146 @@ const BulkActionModal = ({
-//                       'This will unpublish the active locale versions <em>(from Internationalization)</em>',
-//                   },
-//                   {
-//                     em: Emphasis,
-//                   }
-//                 )}
-//               </Typography>
-//             </Box>
-//           )}
-//         </Flex>
-//       ),
-//       confirmButton: formatMessage({
-//         id: 'app.utils.unpublish',
-//         defaultMessage: 'Unpublish',
-//       }),
-//       onConfirm: handleConfirmUnpublishAllData,
-//     },
-//   };
-// };
+const DeleteAction: BulkActionComponent = ({ documents, model }) => {
+  const { formatMessage } = useIntl();
+  const { schema: contentType } = useDoc();
+  const { selectRow } = useTable('deleteAction', (state) => state);
+  const hasI18nEnabled = Boolean(contentType?.pluginOptions?.i18n);
+  const [{ query }] = useQueryParams<{ plugins?: { i18n?: { locale?: string } } }>();
+  const params = React.useMemo(() => buildValidParams(query), [query]);
+  const hasDeletePermission = useDocumentRBAC('deleteAction', (state) => state.canDelete);
+  const { deleteMany: bulkDeleteAction } = useDocumentActions();
+  const documentIds = documents.map(({ documentId }) => documentId);
+
+  const handleConfirmBulkDelete = async () => {
+    const res = await bulkDeleteAction({
+      documentIds,
+      model,
+      params,
+    });
+    if (!('error' in res)) {
+      selectRow([]);
+    }
+  };
+
+  if (!hasDeletePermission) return null;
+
+  return {
+    actionType: 'delete',
+    variant: 'danger-light',
+    label: formatMessage({ id: 'global.delete', defaultMessage: 'Delete' }),
+    dialog: {
+      type: 'dialog',
+      title: formatMessage({
+        id: 'app.components.ConfirmDialog.title',
+        defaultMessage: 'Confirmation',
+      }),
+      content: (
+        <Flex direction="column" alignItems="stretch" gap={2}>
+          <Typography id="confirm-description" textAlign="center">
+            {formatMessage({
+              id: 'popUpWarning.bodyMessage.contentType.delete.all',
+              defaultMessage: 'Are you sure you want to delete these entries?',
+            })}
+          </Typography>
+          {hasI18nEnabled && (
+            <Box textAlign="center" padding={3}>
+              <Typography textColor="danger500">
+                {formatMessage(
+                  {
+                    id: getTranslation('Settings.list.actions.deleteAdditionalInfos'),
+                    defaultMessage:
+                      'This will delete the active locale versions <em>(from Internationalization)</em>',
+                  },
+                  {
+                    em: Emphasis,
+                  }
+                )}
+              </Typography>
+            </Box>
+          )}
+        </Flex>
+      ),
+      onConfirm: handleConfirmBulkDelete,
+    },
+  };
+};
+
+const UnpublishAction: BulkActionComponent = ({ documents, model }) => {
```

Comment:
Shall we add a `type` so people can easily identify them in the reducer? i.e.

```
UnpublishAction.type = "unpublish"
```

Reply:
I added actionType to indentify, as type was used [here](https://github.com/strapi/strapi/blob/fix/bulk-actions-migration/packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/Actions.tsx#L44), couldn't omit it correctly

Reply:
I think there was some confusion how to approach this, check this PR https://github.com/strapi/strapi/pull/20256

It's done the same way we do document actions